.github/workflows/maven-version-determiner.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/python3
+import re
+import subprocess
+import sys
+
+
+def get_arg(arg_idx) -> str:
+    return sys.argv[arg_idx]
+
+
+def get_current_version() -> str:
+    current_version_cmd = subprocess.run(
+        "mvn help:evaluate -Dexpression=project.version -q -DforceStdout",
+        shell=True, capture_output=True, text=True)
+    return current_version_cmd.stdout
+
+
+def get_release_version(release_type: str, current_version: str) -> str:
+    major, minor, patch = determine_new_version(current_version, release_type)
+    return str(major) + "." + str(minor) + "." + str(patch)
+
+
+def get_snapshot_version(release_type: str, current_version: str) -> str:
+    major, minor, patch = determine_new_version(current_version, release_type)
+    patch += 1
+    return str(major) + "." + str(minor) + "." + str(patch) + "-SNAPSHOT"
+
+
+def get_version_tag(release_type: str, current_version: str) -> str:
+    major, minor, patch = determine_new_version(current_version, release_type)
+    return "v" + str(major) + "." + str(minor) + "." + str(patch)
+
+
+def determine_new_version(current_version, release_type):
+    major, minor, patch = dissect_version(current_version)
+
+    match release_type:
+        case "MAJOR":
+            major, minor, patch = get_major_release_version(major)
+        case "MINOR":
+            major, minor, patch = get_minor_release_version(major, minor)
+        case "PATCH":
+            major, minor, patch = get_patch_release_version(major, minor, patch)
+        case _:
+            print("Second arg has to be `MAJOR`, `MINOR` or `PATCH`")
+            sys.exit()
+
+    return major, minor, patch
+
+
+def dissect_version(current_version) -> (int, int, int):
+    version_regex = re.compile(
+        r'^(?P<major>[0-9]+)\.(?P<minor>[0-9+])\.(?P<patch>[0-9+])(-SNAPSHOT)*')
```

Comment:
We might want to use the official regex (even tho it is less readable) as it would be more robust in time: https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string

Reply:
I now use the regex suggested on semver.org. I formatted the regex slightly to increase its readability and added a comment where I took it from.